From 7f5989addebe5d900032a72cf9428bf698665d67 Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Date: Tue, 8 Nov 2022 16:39:42 +0200
Subject: [PATCH 01/13] bitops: Add BIT_32 and BIT_64 macros

When applying some MISRA rules, lots of issues are raised with BIT macro
on AARCH32, and cast on uint32_t would be required (Rule 10.3).
The macros BIT_32 and BIT_64 are then created for 32bit and 64bit.
Then the BIT macro defaults on BIT_64 on AARCH64,
and on BIT_32 on AARCH32.

This is an adaptation of:
'utils: Add BIT_32 and BIT_64 macros' from arm-trusted-firmware
repository.

Upstream-Status: Pending 

Signed-off-by: Yann Gautier <yann.gautier@st.com>
Issue: ALB-8357
Signed-off-by: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 include/linux/bitops.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/linux/bitops.h b/include/linux/bitops.h
index a07c70fd48..f11ca130d4 100644
--- a/include/linux/bitops.h
+++ b/include/linux/bitops.h
@@ -7,6 +7,8 @@
 #include <linux/kernel.h>
 
 #ifdef	__KERNEL__
+#define BIT_32(nr)		(1u << (nr))
+#define BIT_64(nr)		BIT(nr)
 #define BIT(nr)			(1UL << (nr))
 #define BIT_ULL(nr)		(1ULL << (nr))
 #define BIT_MASK(nr)		(1UL << ((nr) % BITS_PER_LONG))
-- 
2.17.1

